pkg/providers/instanceprofile/instanceprofile.go

```diff
@@ -61,8 +60,10 @@ func NewDefaultProvider(region string, iamapi sdk.IAMAPI, cache *cache.Cache) *D
 
 func (p *DefaultProvider) Create(ctx context.Context, m ResourceOwner) (string, error) {
 	profileName := m.InstanceProfileName(options.FromContext(ctx).ClusterName, p.region)
```

Comment:
Should we add a test for this behavior?

Reply:
I am not sure if we can really, since `InstanceProfileName` is an interface method that EC2nodeclass implements. Even if we decide to swap out the entire nodeClass with a testNodeClass, the status controller will not accept it since only expects an EC2nodeclass.